File: backend/load_skool_courses.py
#!/usr/bin/env python3
"""
Load Skool Courses from CSV into MongoDB
Imports course and lesson data from Skool export CSV file
"""

import csv
import re
import asyncio
from motor.motor_asyncio import AsyncIOMotorClient
from datetime import datetime
from bson.objectid import ObjectId
import uuid

# Database configuration
MONGO_URL = "mongodb://admin:vcsa_local_dev_2024@localhost:27019"
DB_NAME = "vcsa"

# CSV file path
CSV_FILE = "wiki/skool/_Vacation Club Sales Academy _ Skool - Hoja 1.csv"

def convert_youtube_to_embed(url):
    """Convert YouTube watch URL to embed format"""
    if not url:
        return ""

    # Handle different YouTube URL formats
    patterns = [
        r'(?:https?:\/\/)?(?:www\.)?youtube\.com\/watch\?v=([^&]+)',
        r'(?:https?:\/\/)?(?:www\.)?youtube\.com\/embed\/([^/?]+)',
        r'(?:https?:\/\/)?youtu\.be\/([^/?]+)',
    ]

    for pattern in patterns:
        match = re.search(pattern, url)
        if match:
            video_id = match.group(1)
            return f"https://www.youtube.com/embed/{video_id}"

    # If URL doesn't match known patterns, return as-is
    return url

def estimate_duration_from_url(url):
    """Estimate video duration in minutes (default for now)"""
    # In production, you could use YouTube API to get actual duration
    return 15  # Default 15 minutes per video

async def load_skool_courses():
    """Load courses and lessons from CSV into MongoDB"""

    # Connect to MongoDB
    client = AsyncIOMotorClient(MONGO_URL)
    db = client[DB_NAME]

    print("📚 Loading Skool Courses from CSV...")
    print(f"📁 CSV File: {CSV_FILE}")
    print(f"🗄️  Database: {DB_NAME}")
    print("")

    # Read CSV and organize by module
    courses_data = {}

    with open(CSV_FILE, 'r', encoding='utf-8') as file:
        csv_reader = csv.DictReader(file)

        for row in csv_reader:
            module = row['Modulo'].strip()
            title = row['Titulo'].strip()
            copy = row['Copy'].strip() if row['Copy'] else ""
            video_url = row['Link'].strip()

            if not module or not title:
                continue

            # Initialize module if not exists
            if module not in courses_data:
                courses_data[module] = {
                    'title': module,
                    'description': f"Training content from Skool: {module}",
                    'lessons': []
                }

            # Add lesson to module
            courses_data[module]['lessons'].append({
                'title': title,
                'description': copy or f"Learn about {title}",
                'video_url': video_url,
                'order': len(courses_data[module]['lessons']) + 1
            })

    print(f"📊 Found {len(courses_data)} courses with {sum(len(c['lessons']) for c in courses_data.values())} total lessons")
    print("")

    # Load courses into database
    admin_user = await db.users.find_one({"role": "admin"})
    if not admin_user:
        admin_user = await db.users.find_one({})  # Use first user if no admin

    created_by = admin_user['user_id'] if admin_user else 'system'

    for module_name, course_data in courses_data.items():
        print(f"📖 Processing Course: {module_name}")

        # Check if course already exists
        existing_course = await db.courses.find_one({"title": module_name})

        if existing_course:
            print(f"   ✅ Course already exists: {existing_course['course_id']}")
            course_id = existing_course['course_id']

            # Delete existing lessons for this course to avoid duplicates
            await db.lessons.delete_many({"course_id": course_id})
            print(f"   🗑️  Deleted {len(course_data['lessons'])} old lessons")
        else:
            # Create new course
            course_id = f"course_{uuid.uuid4().hex[:12]}"

            course_doc = {
                "course_id": course_id,
                "title": course_data['title'],
                "description": course_data['description'],
                "thumbnail": "https://img.youtube.com/vi/default/maxresdefault.jpg",  # Generic thumbnail
                "category": "masterclass",  # Skool content as masterclasses
                "min_level": 1,
                "vip_only": False,
                "lessons": [],
                "created_at": datetime.now(),
                "created_by": created_by
            }

            await db.courses.insert_one(course_doc)
            print(f"   ✅ Created course: {course_id}")

        # Create lessons
        lesson_count = 0
        for lesson_data in course_data['lessons']:
            lesson_id = f"lesson_{uuid.uuid4().hex[:12]}"

            # Convert YouTube URL to embed format
            embed_url = convert_youtube_to_embed(lesson_data['video_url'])
            duration = estimate_duration_from_url(lesson_data['video_url'])

            lesson_doc = {
                "lesson_id": lesson_id,
                "course_id": course_id,
                "title": lesson_data['title'],
                "description": lesson_data['description'],
                "video_url": embed_url,
                "duration": duration,
                "order": lesson_data['order'],
                "created_at": datetime.now()
            }

            await db.lessons.insert_one(lesson_doc)
            lesson_count += 1

            print(f"   📹 Lesson {lesson_count}: {lesson_data['title']}")
            print(f"      URL: {embed_url}")

        # Update course with lesson IDs
        lesson_ids = [f"lesson_{uuid.uuid4().hex[:12]}" for _ in range(lesson_count)]
        await db.courses.update_one(
            {"course_id": course_id},
            {"$set": {"lessons": lesson_ids}}
        )

        print(f"   ✅ Created {lesson_count} lessons")
        print("")

    # Summary
    total_courses = len(courses_data)
    total_lessons = sum(len(c['lessons']) for c in courses_data.values())

    print("=" * 60)
    print("📊 SUMMARY")
    print("=" * 60)
    print(f"✅ Courses processed: {total_courses}")
    print(f"✅ Lessons created: {total_lessons}")
    print(f"✅ Database: {DB_NAME}")
    print("")

    # Verify data
    print("🔍 VERIFICATION")
    print("=" * 60)

    courses_count = await db.courses.count_documents({})
    lessons_count = await db.lessons.count_documents({})

    print(f"📚 Total courses in database: {courses_count}")
    print(f"📹 Total lessons in database: {lessons_count}")
    print("")

    # List all courses
    print("📋 COURSES LIST:")
    print("-" * 60)

    courses = await db.courses.find({}, {"_id": 0}).to_list(50)
    for course in courses:
        lesson_count = await db.lessons.count_documents({"course_id": course['course_id']})
        print(f"📖 {course['title']}")
        print(f"   ID: {course['course_id']}")
        print(f"   Category: {course.get('category', 'N/A')}")
        print(f"   Lessons: {lesson_count}")
        print(f"   VIP Only: {course.get('vip_only', False)}")
        print("")

    client.close()
    print("✅ Skool courses loaded successfully!")

if __name__ == "__main__":
    asyncio.run(load_skool_courses())

File: vcsAcademy-detailed-clean/backend/load_skool_courses.js
// Load Skool Courses from CSV into MongoDB
// Run with: mongosh mongodb://admin:vcsa_local_dev_2024@localhost:27019/vcsa load_skool_courses.js

// Course data from CSV
const skoolCourses = [
  {
    title: "FREE RESOURCES (The RoadMAP 2026)",
    description: "Essential training resources for vacation club sales professionals",
    category: "masterclass",
    lessons: [
      {
        title: "Breaking The Pact",
        description: "Learn how to break traditional sales patterns",
        video_url: "https://www.youtube.com/embed/yN3lahhU-4c",
        duration: 15,
        order: 1
      },
      {
        title: "First Visit Incentives",
        description: "Master first visit incentive strategies",
        video_url: "https://www.youtube.com/embed/IZFrfqD6aBY",
        duration: 15,
        order: 2
      },
      {
        title: "The Residence Story",
        description: "Learn to tell compelling residence stories",
        video_url: "https://www.youtube.com/embed/74LcxFvsMHI",
        duration: 15,
        order: 3
      },
      {
        title: "The Concept Pitch",
        description: "Perfect your concept presentation skills",
        video_url: "https://www.youtube.com/embed/zkOG6Eyi9Cc",
        duration: 15,
        order: 4
      },
      {
        title: "No Comes at a Price",
        description: "Handle price objections effectively",
        video_url: "https://www.youtube.com/embed/oOrz6H7XSvU",
        duration: 15,
        order: 5
      }
    ]
  },
  {
    title: "PART 1 of The FRONT TO BACK CHALLENGE",
    description: "Front to Back Challenge - Complete sales process mastery",
    category: "workshop",
    lessons: [
      {
        title: "FRONT TO BACK CHALLENGE",
        description: "Master the complete sales process from front to back",
        video_url: "https://www.youtube.com/embed/HmZPlXY6Dqk",
        duration: 20,
        order: 1
      }
    ]
  }
];

// Get admin user (created_by)
const adminUser = db.users.findOne({role: "admin"}) || db.users.findOne({});
const createdBy = adminUser ? adminUser.user_id : 'system';

print("📚 Loading Skool Courses...");
print("");

let totalCoursesCreated = 0;
let totalLessonsCreated = 0;

skoolCourses.forEach((courseData, index) => {
  print(`📖 Processing Course ${index + 1}: ${courseData.title}`);

  // Check if course already exists
  const existingCourse = db.courses.findOne({title: courseData.title});

  let courseId;

  if (existingCourse) {
    print(`   ✅ Course already exists: ${existingCourse.course_id}`);
    courseId = existingCourse.course_id;

    // Delete existing lessons
    const deletedLessons = db.lessons.deleteMany({course_id: courseId});
    print(`   🗑️  Deleted ${deletedLessons.deletedCount} old lessons`);
  } else {
    // Create new course
    courseId = `course_${Date.now().toString(36)}_${index}`;

    const courseDoc = {
      course_id: courseId,
      title: courseData.title,
      description: courseData.description,
      thumbnail: "https://img.youtube.com/vi/default/maxresdefault.jpg",
      category: courseData.category,
      min_level: 1,
      vip_only: false,
      lessons: [],
      created_at: new Date(),
      created_by: createdBy
    };

    db.courses.insertOne(courseDoc);
    print(`   ✅ Created course: ${courseId}`);
  }

  // Create lessons
  const lessonIds = [];

  courseData.lessons.forEach((lessonData, lessonIndex) => {
    const lessonId = `lesson_${Date.now().toString(36)}_${index}_${lessonIndex}`;

    const lessonDoc = {
      lesson_id: lessonId,
      course_id: courseId,
      title: lessonData.title,
      description: lessonData.description,
      video_url: lessonData.video_url,
      duration: lessonData.duration,
      order: lessonData.order,
      created_at: new Date()
    };

    db.lessons.insertOne(lessonDoc);
    lessonIds.push(lessonId);

    print(`   📹 Lesson ${lessonIndex + 1}: ${lessonData.title}`);
    print(`      URL: ${lessonData.video_url}`);

    totalLessonsCreated++;
  });

  // Update course with lesson IDs
  db.courses.updateOne(
    {course_id: courseId},
    {$set: {lessons: lessonIds}}
  );

  print(`   ✅ Created ${lessonIds.length} lessons`);
  print("");

  totalCoursesCreated++;
});

// Summary
print("=" * 60);
print("📊 SUMMARY");
print("=" * 60);
print(`✅ Courses processed: ${totalCoursesCreated}`);
print(`✅ Lessons created: ${totalLessonsCreated}`);
print("");

// Verification
print("🔍 VERIFICATION");
print("=" * 60);

const coursesCount = db.courses.countDocuments({});
const lessonsCount = db.lessons.countDocuments({});

print(`📚 Total courses in database: ${coursesCount}`);
print(`📹 Total lessons in database: ${lessonsCount}`);
print("");

// List all courses
print("📋 COURSES LIST:");
print("-" * 60);

const allCourses = db.courses.find({}, {_id: 0}).toArray();
allCourses.forEach(course => {
  const courseLessonsCount = db.lessons.countDocuments({course_id: course.course_id});
  print(`📖 ${course.title}`);
  print(`   ID: ${course.course_id}`);
  print(`   Category: ${course.category || 'N/A'}`);
  print(`   Lessons: ${courseLessonsCount}`);
  print(`   VIP Only: ${course.vip_only || false}`);
  print("");
});

print("✅ Skool courses loaded successfully!");
